instrumentation/sofa-rpc/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/sofa/rpc/SofaRpcTelemetryBuilder.java

```diff
@@ -0,0 +1,127 @@
+
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.instrumentation.api.incubator.semconv.rpc.RpcClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.incubator.semconv.rpc.RpcClientMetrics;
+import io.opentelemetry.instrumentation.api.incubator.semconv.rpc.RpcServerAttributesExtractor;
+import io.opentelemetry.instrumentation.api.incubator.semconv.rpc.RpcServerMetrics;
+import io.opentelemetry.instrumentation.api.incubator.semconv.rpc.RpcSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.semconv.network.NetworkAttributesExtractor;
+import io.opentelemetry.instrumentation.api.semconv.network.ServerAttributesExtractor;
+import io.opentelemetry.instrumentation.sofa.rpc.internal.SofaRpcClientNetworkAttributesGetter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.UnaryOperator;
+import javax.annotation.Nullable;
+
+/** A builder of {@link SofaRpcTelemetry}. */
+public final class SofaRpcTelemetryBuilder {
+
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.sofa-rpc";
+
+  // copied from PeerIncubatingAttributes
+  private static final AttributeKey<String> PEER_SERVICE = AttributeKey.stringKey("peer.service");
+
+  private final OpenTelemetry openTelemetry;
+  @Nullable private String peerService;
+  private final List<AttributesExtractor<SofaRpcRequest, SofaResponse>> attributesExtractors =
+      new ArrayList<>();
+  private UnaryOperator<SpanNameExtractor<SofaRpcRequest>> clientSpanNameExtractorTransformer =
+      UnaryOperator.identity();
+  private UnaryOperator<SpanNameExtractor<SofaRpcRequest>> serverSpanNameExtractorTransformer =
+      UnaryOperator.identity();
+
+  SofaRpcTelemetryBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = openTelemetry;
+  }
+
+  /** Sets the {@code peer.service} attribute for RPC client spans. */
+  public void setPeerService(String peerService) {
+    this.peerService = peerService;
+  }
+
+  /**
+   * Adds an additional {@link AttributesExtractor} to invoke to set attributes to instrumented
+   * items.
+   */
+  @CanIgnoreReturnValue
+  public SofaRpcTelemetryBuilder addAttributesExtractor(
+      AttributesExtractor<SofaRpcRequest, SofaResponse> attributesExtractor) {
+    attributesExtractors.add(attributesExtractor);
+    return this;
+  }
+
+  /** Sets custom client {@link SpanNameExtractor} via transform function. */
+  @CanIgnoreReturnValue
+  public SofaRpcTelemetryBuilder setClientSpanNameExtractor(
```

Comment:
```suggestion
  public SofaRpcTelemetryBuilder setClientSpanNameExtractorCustomizer(
```
it needs to following latest naming convention, https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/15529